pubtools/pulplib/_impl/fake/client.py

```diff
@@ -93,6 +93,31 @@ def search_repository(self, criteria=None):
         random.shuffle(repos)
         return self._prepare_pages(repos)
 
+    def search_content(self, criteria=None):
+        criteria = criteria or Criteria.true()
+        out = []
+
+        # Pass the criteria through the code used by the real client to build
+        # up the Pulp query. We don't actually *use* the resulting query since
+        # we're not accessing a real Pulp server. The point is to ensure the
+        # same validation and error behavior as used by the real client also
+        # applies to the fake.
+        prepared_search = search_for_criteria(criteria, Unit)
+
```

Comment:
The real client will raise an error somewhere around here, if the criteria included some type which is not supported by the server.

Could this client please do the same thing?  Otherwise, tests using the fake client won't be able to find bugs such as a search query including some useless wrong type id.

Reply:
I don't think the implemented logic is correct, looks like it's going to raise errors if you try to search on a content type for which there's no data.  No data doesn't mean the type ID isn't supported.

Seems like the list of types could be gathered from the classes which exist in this library, or even just set as a hardcoded list.